packages/components/src/utils/get-focusable-children.ts

```diff
@@ -22,8 +22,18 @@ const focusDisablingSelector = `:where(${[
   'details:not([open]) > *:not(details > summary:first-of-type)',
   'details:not([open]) > *:not(details > summary:first-of-type) *',
   '[tabindex^="-"]',
+  '[hidden]:not([hidden="false"])',
 ].join(',')})`;
 
-export function getFocusableChildren(element: Element): NodeListOf<HTMLElement> {
-  return element.querySelectorAll(`& > ${focusableSelector}:not(${focusDisablingSelector})`);
+export function getFocusableChildren(element: Element): HTMLElement[] {
+  const focusableChildren = element.querySelectorAll(
+    `${focusableSelector}:not(${focusDisablingSelector})`,
+  );
+
+  const visibleFocusableChildren = Array.from(focusableChildren).filter(child => {
+    const style = window.getComputedStyle(child.parentElement);
+    return style.display !== 'none' && style.visibility !== 'hidden';
+  });
+
+  return visibleFocusableChildren as HTMLElement[];
```

Comment:
With `as` you are forcing the type, this is not a good practice and should be avoided as much as possible:
```suggestion
  return visibleFocusableChildren;
```